tools/kwiver_tool_runner.cxx

```diff
@@ -181,13 +184,27 @@ int main(int argc, char *argv[])
   // Global shared context
   // Allocated on the stack so it will automatically clean up
   //
+  using kvpf = kwiver::vital::plugin_factory;
+
   applet_context_t tool_context = std::make_shared< kwiver::tools::applet_context >();
 
   kwiver::vital::plugin_manager& vpm = kwiver::vital::plugin_manager::instance();
   const std::string exec_path = kwiver::vital::get_executable_path();
   vpm.add_search_path(exec_path + "/../lib/kwiver/modules");
```

Comment:
So... my original thought for this feature was not to load the plugin, ask what type it is, and unload it if not the right type (which I suspect is actually *worse*), but to only search the subdirectories of the plugin type(s) requested.

Reply:
It is hard to determine which subdirectories to search if the list of sub directories is user supplied.

Reply:
So... the way Qt gets around this problem (and to an extent, ViViA also) is you don't supply it the path that actually contains the `.dll`s/`.so`s, you supply it the path that contains the well-known *subdirectories* for each type of module.

In this case, the least invasive way to implement that would be by removing "modules/applets" from the explicitly added paths and instead searching the "applets" subdirectory of each path in the list *iff* applets should be loaded in addition to regular modules.